django/urls/base.py

```diff
@@ -85,7 +94,16 @@ def reverse(viewname, urlconf=None, args=None, kwargs=None, current_app=None):
                 ns_pattern, resolver, tuple(ns_converters.items())
             )
 
-    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
+    resolved_url = resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
+    if query:
+        if isinstance(query, QueryDict):
+            query_string = query.urlencode()
+        else:
+            query_string = urlencode(query, doseq=True)
+        resolved_url += "?" + query_string
+    if fragment:
```

Comment:
```suggestion
    if fragment is not None:
```
Do we want to accept an empty fragment? (and add a test for `fragment=""`)?